### DISCUSSION TITLE
dynamic fields based on a select option nested layout

### DISCUSSION THREAD

**Discussion by T3mmyKay** [filamentphp/filament#13044] [Help]
### Package

Panel builder

### Package Version

v3.2.0

### How can we help you?

Forms\Components\Section::make()
                   ->schema([
                            Select::make('tax_id')
                                ->relationship('tax')
                                ->getOptionLabelFromRecordUsing(fn(Tax $record) => "{$record->name}@{$record->rate}%")
                                ->placeholder('None')
                                ->default('')
                                ->label('Applicable Tax')
                                ->native(false),
                            Select::make('tax_type')
                                ->options(TaxTypeEnum::class)
                                ->native(false)
                                ->default(TaxTypeEnum::INCLUSIVE)
                                ->hintIcon('heroicon-m-question-mark-circle', tooltip: str('Inclusive includes the tax in the product price, while exclusive, adds tax on top of the price')->inlineMarkdown()->toHtmlString())
                                ->live()
                                ->afterStateUpdated(function (Select $component) {
                                    return $component
                                        ->getContainer()
                                        ->getComponent('variationGroup')
                                        ->getContainer()
                                        ->getComponent('dynamicTypeFields')
                                        ->getChildComponentContainer()
                                        ->fill();
                                })
                                ->hintColor('primary'),
                                
   This is the select, but I want I want to fill is in another group
   
   Forms\Components\Group::make()
                           ->columnSpanFull()
                           ->key('variationGroup')
                           ->relationship('variation')
                           ->label('Purchase Price')
                           ->columns([
                               'default' => 1,
                               'sm' => 2,
                               'md' => 4
                           ])->schema([
                               TextInput::make('default_purchase_price')
                                   ->numeric()
                                   ->label('Purchase Price')
                                   ->hint('Exc tax')
                                   ->placeholder('0.00'),
                               TextInput::make('dpp_inc_tax')
                                   ->numeric()
                                   ->label('Purchase Price')
                                   ->hint('Inc tax')
                                   ->placeholder('0.00'),
                               TextInput::make('profit_percent')
                                   ->numeric()
                                   ->label('Profit Margin')
                                   ->default(25)
                                   ->hintIcon(icon: 'heroicon-m-question-mark-circle', tooltip: 'Profit margin in percentage')
                                   ->hintColor('primary')
                                   ->placeholder('0.00'),
                               Forms\Components\Group::make()->schema(fn(Get $get): array => match ($get('tax_type')) {
                                   TaxTypeEnum::INCLUSIVE => [
                                       TextInput::make('employee_number')
                                           ->required(),
                                   ],
                                   TaxTypeEnum::EXCLUSIVE => [
                                       TextInput::make('daily_rate')
                                           ->numeric()
                                           ->required()
                                           ->prefix('€'),],
                                   default => [],
                               })->key('dynamicTypeFields')
                           ]),
Please how can I make it work

**Comment by leandrocfe** [comment]
Sorry, what is the question? Could you explain what are you trying to do?

**Comment by T3mmyKay** [comment]
I wanted to generate a dynamic field based on the selected tax type. I saw on the documentation and tried to follow what is there but it wasn't working
